src/mainClass.js

```diff
@@ -2,44 +2,76 @@ import { getData, getAllCategories } from "./source";
 
 export class AllItems {
 
-  constructor(searchCategory, searchLinks, displayLinksForChosenCategory, searchChosenLinksOnly) {
+  constructor(searchCategory, searchLinks, displayLinksForChosenCategory) {
     this.categories = [];
     this.onReceivedCategories = searchCategory;
     this.displayChosenLinks = displayLinksForChosenCategory;
     this.links = [];
     this.onReceivedLinks = searchLinks;
-    this.chosenLinks = [];
-    this.onReceivedChosenLinks = searchChosenLinksOnly;
+    this.storageLinks = JSON.parse(localStorage.getItem('links'));
+    this.storageCategories = JSON.parse(localStorage.getItem('categories'));
+    this.filtered;
   }
 
-  setOnCategoriesReceived(onReceivedCategories) { 
-    this.onReceivedCategories = onReceivedCategories 
+  setOnCategoriesReceived(onReceivedCategories) {
+    this.onReceivedCategories = onReceivedCategories
   }
 
   setOnLinksReceived(onReceivedLinks) {
     this.onReceivedLinks = onReceivedLinks;
   }
 
-  setOnChosenLinksReceived(onReceivedChosenLinks) {
-    this.onReceivedChosenLinks = onReceivedChosenLinks;
-  }
-
   setDisplayChosenLinks(displayChosenLinks) {
     this.displayChosenLinks = displayChosenLinks;
   }
 
+  updateStorageLinks() {
+    localStorage.setItem('links', JSON.stringify(this.links));
+  }
+
+  updateStorageCategories() {
+    localStorage.setItem('categories', JSON.stringify(this.categories));
+  }
+
+  markAsImportant(el) {
+    if (el.important === false) {
+      el.important = true;
+    } else {
+      el.important = false;
+    }
+  }
+
+  setFiltered(param) {
+    this.filtered = param;
+  }
+
+  getAllLinks(filtered, e) {
+    if (filtered !== null) {
+      let filteredCategoriesAfterSearch = this.links.filter(el => {
+        if (el.API.toLowerCase().includes(e.target.value.toLowerCase())) {
```

Comment:
this should be filtered from filtered prop, not from received prop. Input should set prop, and call this method.